### PR TITLE
database integraion added

### PR DESCRIPTION
**Pull request by dmitrii-esin**
# All tasks implemented without additional stocks table:

## Main:
- [x] 1 - Task 4.1 is implemented
- [x] 3 - TASK 4.2 is implemented lambda links are provided and returns data
- [x] 4 - TASK 4.3 is implemented lambda links are provided and products is stored in DB (call TASK 4.2 to see the product)
- [x] 5 - Your own Frontend application is integrated with product service (/products API) and products from product-service are represented on Frontend. Link to a working Front-End application is provided for cross-check reviewer.

## Additional:
- [x] +1 - POST/products lambda functions returns error 400 status code if product data is invalid
- [x] +1 - All lambdas return error 500 status code on any error (DB connection, any unhandled error in code)
- [x] +1 - All lambdas do console.log for each incoming requests and their arguments
- [ ] +1 - Transaction based creation of product (in case stock creation is failed then related to this stock product is not created and not ready to be used by the end user and vice versa) (https://devcenter.kinvey.com/nodejs/tutorials/bl-transactional-support)

## endpoints:
  GET - https://ie2svy46v9.execute-api.eu-west-1.amazonaws.com/dev/products
  GET - https://ie2svy46v9.execute-api.eu-west-1.amazonaws.com/dev/products/{id}
  POST - https://ie2svy46v9.execute-api.eu-west-1.amazonaws.com/dev/products
  GET - https://ie2svy46v9.execute-api.eu-west-1.amazonaws.com/dev/migrate

## frontend app link: 
  https://d12t0bvcb8pyyn.cloudfront.net

### CODE REVIEW

File: product-service/src/functions/migrateSchema/handler.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  formatErrorResponse,
+} from "@libs/apiResponseBuilder";
+
+export const migrateSchema = (client: Client) => async (event, _context) => {
+  try {
+    await client.query('CREATE EXTENSION IF NOT EXISTS "uuid-ossp";');
+    await client.query(`
+            CREATE TABLE IF NOT EXISTS products (
+                id UUID PRIMARY KEY DEFAULT uuid_generate_v4(),
+                count INT,
```
Comment (artem-blazhko) [inline product-service/src/functions/migrateSchema/handler.ts:16]: you should have separated table that will contain count and product id

File: product-service/src/functions/migrateSchema/handler.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                date TEXT,
+                description TEXT,
+                image TEXT,
+                location TEXT,
+                price INT,
+                title TEXT
```
Comment (artem-blazhko) [inline product-service/src/functions/migrateSchema/handler.ts:22]: Title should be **not null**
image

File: product-service/src/services/postgres-memory-product-service.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      const query = {
+        text: `INSERT INTO ${this.tableName}(count, description, date, location, price, title, image) VALUES($1, $2, $3, $4, $5, $6, $7) RETURNING *`,
+        values: [
+          product.count,
+          product.description,
+          product.date,
+          product.location,
+          product.price,
+          product.title,
+          product.image,
```
Comment (artem-blazhko) [inline product-service/src/services/postgres-memory-product-service.ts:61]: You should store data in 2 different tables

**Review by artem-blazhko — COMMENTED** [review]
After adding a new product I see very strange view. Product doesn't have title and prise
image

But I sent the correct body and received object will nulls
image

**Comment by dmitrii-esin** [comment]
> After adding a new product I see very strange view. Product doesn't have title and prise
> image
> 
> But I sent the correct body and received object will nulls
> image

Thank you for your attention! 
I fixed that.

**Pull request merged**